src/services/opensea.js

```diff
@@ -34,23 +39,61 @@ const requestConfig = {
   },
 };
 
-export const getLimitedData =
-  (url: string, data: Array<Object>, limit: number, offset: number,
-    responseDataKey: string, resolve: Function, reject: Function) => {
-    httpRequest.get(`${url}&limit=${limit}&offset=${offset}`, requestConfig)
-      .then(({ data: responseData }) => {
-        const retrievedData = data.concat(responseData[responseDataKey]);
-        const newOffset = offset + limit;
-        if (responseData[responseDataKey].length === limit) {
-          setTimeout(() => {
-            getLimitedData(`${url}&limit=${limit}&offset=${newOffset}`,
-              retrievedData, limit, newOffset, responseDataKey, resolve, reject);
-          }, 1000);
-        } else {
-          setTimeout(() => {
-            resolve(retrievedData);
-          }, 1000);
-        }
-      })
-      .catch(error => { reject(error); });
-  };
+const paginatedRequest = (
+  url: string,
+  paginatedData: Array<Object> = [],
+  limit: number = 300,
+  offset: number = 0,
+) => {
+  return new Promise(async (resolve) => {
+    const { data: responseData } = await httpRequest.get(`${url}&limit=${limit}&offset=${offset}`, requestConfig);
+    const assets = responseData?.assets || [];
+    const updatedPaginatedData = [...paginatedData, ...assets];
+    const newOffset = offset + limit;
+    if (assets.length === limit) {
+      setTimeout(() => { paginatedRequest(url, updatedPaginatedData, limit, newOffset); }, 1000);
```

Comment:
Why use 1sec delay?

Reply:
Something that was ported from previous implementation, however, noticed that this is incorrect in any way.

1s delay was probably added to avoid request throttling, however, removed it altogether with a fix for this.